sdk/typescript/src/types/sui-bcs.ts

```diff
@@ -292,4 +313,53 @@ bcs.registerStructType('TransactionData', {
   gasBudget: 'u64',
 });
 
+// ========== Deprecated ===========
```

Comment:
@666lcz I'm not super happy about this pattern, but in order to serialize the old protocol, I needed to register a whole new sub-tree of types, suffixed with `_Deprecated`.  I couldn't just add a new variant to `ObjectArg` because its variant index in BCS wouldn't match.

The ideal solution here would be to to create two instances of `bcs` with different types registered.  I think that's worth doing if this kind of pattern is going to be common, but is out of scope of this PR -- it requires BCS to no longer expose a singleton API (i.e. all static functions) which is a breaking change for that package.